bot/exts/evergreen/minesweeper.py

```diff
@@ -278,6 +290,14 @@ def reveal_zeros(self, revealed: GameBoard, board: GameBoard, x: int, y: int) ->
             await game.chat_msg.edit(content=new_msg)
         del self.games[ctx.author.id]
 
+    @reveal_command.error
+    @end_command.error
+    @flag_command.error
+    async def user_not_playing_handler(self, ctx: commands.Context, error: Exception) -> None:
+        """Handle `KeyError` that is raised in reveal or end command when getting game."""
+        if isinstance(error, UserNotPlayingError):
+            await ctx.send("Game don't exist.")
```

Comment:
Exceptions raised within a `d.py` command will be wrapped in [`CommandInvokeError`](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.CommandInvokeError) which carries on `original` attribute with the specific exception. As a result, this doesn't catch the `UserNotPlayingError`, as `error` isn't an instance thereof - `error.original` is.

The error handler cog gets around this via [safely getting the original](https://github.com/python-discord/seasonalbot/blob/38d0098b0f2dce38e032be235b6dd2f1e2109b4a/bot/exts/evergreen/error_handler.py#L49) before processing it.

Additionally, as *all* errors from the commands will end up here, we should also make sure they aren't silenced. Let's [log the exception](https://github.com/python-discord/seasonalbot/blob/38d0098b0f2dce38e032be235b6dd2f1e2109b4a/bot/exts/evergreen/error_handler.py#L124) just like the error handler does.

Finally, since `user_not_playing_handler` will receive all errors, not just `UserNotPlayingError`, it would be better to just name it `error_handler`, or similar. The docstring still mentions `KeyError` which no longer holds.

---

An alternative to re-implementing the error handler cog's logic here, we could also consider moving our logic there. If you decide to do this, please move the `UserNotPlayingError` definition to `bot.utils.exceptions` - importing Exceptions classes across `d.py` extension modules is dangerous and can lead to unexpected behaviour, as the definition may be executed multiple times.